File: C/01TwoSum/twosum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        direct = {};
        
        for (key,value) in enumerate(nums):
            if target - value in direct:
                return key,direct[target - value]
            direct[value] = key  

File: Go/01_TwoSum/twosum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {

	maps := map[int]int{}

	for i, v := range nums{
		if mi, ok := maps[target - v]; ok {
			return []int{mi, i}
		}
		maps[v] = i
	}

	return nil
}

func main(){

	//num := []int{2, 7, 11, 12}
	num := []int{2, 7, 11, 12}

	data := twoSum(num, 18)

	fmt.Println(data)
}